metadata-ingestion/src/datahub/sdk/lineage_client.py

```diff
@@ -227,9 +240,140 @@ def add_dataset_transform_lineage(
             raise SdkUsageError(
                 f"Dataset {updater.urn} does not exist, and hence cannot be updated."
             )
+
         mcps: List[
-            Union[MetadataChangeProposalWrapper, models.MetadataChangeProposalClass]
+            Union[MetadataChangeProposalWrapper, MetadataChangeProposalClass]
         ] = list(updater.build())
         if query_entity:
             mcps.extend(query_entity)
         self._client._graph.emit_mcps(mcps)
+
+    def add_dataset_lineage_from_sql(
+        self,
+        *,
+        query_text: str,
+        platform: str,
+        platform_instance: Optional[str] = None,
+        env: str = "PROD",
+        default_db: Optional[str] = None,
+        default_schema: Optional[str] = None,
+        schema_aware: bool = True,
+    ) -> None:
+        """Add lineage by parsing a SQL query."""
+        from datahub.sql_parsing.sqlglot_lineage import create_lineage_sql_parsed_result
+
+        # Parse the SQL query to extract lineage information
+        parsed_result = create_lineage_sql_parsed_result(
+            query=query_text,
+            default_db=default_db or "",
+            default_schema=default_schema or "",
+            platform=platform,
+            platform_instance=platform_instance,
+            env=env,
+            graph=self._client._graph,
+            schema_aware=schema_aware,
+        )
+
+        # Validate parsing result
+        if not parsed_result:
+            raise SdkUsageError("Failed to parse SQL query: Unable to parse")
+        if parsed_result.debug_info.error:
```

Comment:
we should distinguish between table errors and column errors - only the former should be fatal and the latter should be a warning